### ISSUE TITLE
Not working with HTTP GitHub API

### ISSUE THREAD

**Issue by bopoda** [jenkinsci/violation-comments-to-github-plugin#32]
Jenkins ver. 2.46.2
Violation Comments to GitHub plugin ver. 1.50 (and v.1.8 is checked as well)

Here the Jenkins output:
```
openjdk version "1.8.0_171"
OpenJDK Runtime Environment (build 1.8.0_171-8u171-b11-2~14.04-b11)
OpenJDK 64-Bit Server VM (build 25.171-b11, mixed mode)
---
--- Jenkins Violation Comments to GitHub ---
---
gitHubUrl: http://github.<company>.com/api/v3/
repositoryOwner: <company>
repositoryName: email-service
pullRequestId: 46
usernamePasswordCredentialsId: false
username: true
password: true
useOAuth2TokenCredentials: true
oAuth2Token: true
createSingleFileComments: false
createCommentWithAllSingleFileComments: false
commentOnlyChangedContent: false
minSeverity: INFO
keepOldComments: false
Running Jenkins Violation Comments To GitHub
PR 46
Workspace: /home/jenkins-slave/workspace/email-service-pull-request
Using OAuth2Token
PR: <company>/email-service/46 on http://github.<company>.com/api/v3/
java.lang.RuntimeException: java.net.ConnectException: Connection refused (Connection refused)
	at se.bjurr.violations.comments.github.lib.GitHubCommentsProvider.<init>(GitHubCommentsProvider.java:59)
	at se.bjurr.violations.comments.github.lib.ViolationCommentsToGitHubApi.toPullRequest(ViolationCommentsToGitHubApi.java:136)
	at org.jenkinsci.plugins.jvctg.perform.JvctgPerformer.doPerform(JvctgPerformer.java:137)
	at org.jenkinsci.plugins.jvctg.perform.JvctgPerformer$1.invoke(JvctgPerformer.java:223)
	at org.jenkinsci.plugins.jvctg.perform.JvctgPerformer$1.invoke(JvctgPerformer.java:211)
	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2731)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:336)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:350)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:673)
	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:347)
	at org.eclipse.egit.github.core.client.GitHubClient.get(GitHubClient.java:733)
	at org.eclipse.egit.github.core.client.PageIterator.next(PageIterator.java:173)
	at org.eclipse.egit.github.core.service.GitHubService.getAll(GitHubService.java:151)
	at org.eclipse.egit.github.core.service.GitHubService.getAll(GitHubService.java:135)
	at org.eclipse.egit.github.core.service.PullRequestService.getCommits(PullRequestService.java:319)
	at se.bjurr.violations.comments.github.lib.GitHubCommentsProvider.<init>(GitHubCommentsProvider.java:56)
	... 13 more
```

I use **enterprise GitHub** repository.
The GitHub repository is working using **HTTP**, not **https**. It is under VPN but the access is granted for Jenkins machine (whitelisted).

I fill Base URL in the settings of plugin:
`http://github.<company>.com/api/v3/`

As you see in the log above, I see the error during the build process:
```
java.lang.RuntimeException: java.net.ConnectException: Connection refused (Connection refused)
```
The same error I see if I send curl request to https endpoint.
```
$ curl -v https://github.<company>.com/api/v3/
*   Trying 10.201.1.201...
* TCP_NODELAY set
* connect to 10.201.1.201 port 443 failed: Connection refused
* Failed to connect to github.<company>.com port 443: Connection refused
* Closing connection 0
curl: (7) Failed to connect to github.<company>.com port 443: Connection refused
```

To conclude:
I expect that Jenkins will send requests to Github using http, not https (as I fill http port in Base URL). But, according to jenkins logs, the https client is used - `at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)`.

**Comment by tomasbjerre** [comment]
You are most likely right about that. Because it crates a GitHub client here: https://github.com/eclipse/egit-github/blob/master/org.eclipse.egit.github.core/src/org/eclipse/egit/github/core/client/GitHubClient.java#L70 and that will always use HTTPS.

Will have a look later.

**Comment by bopoda** [comment]
@tomasbjerre Thanks for the fast reply!

**Comment by tomasbjerre** [comment]
Corrected in 1.83. Open issue again if not working!

https://github.com/tomasbjerre/violation-comments-to-github-lib/commit/30a2768b1a76ab1230156c38bbe27e30ef131b55

**Comment by bopoda** [comment]
verified v.1.83 using HTTP GitHub API.
Well done!
